lib/internal/url.js

```diff
@@ -456,6 +159,352 @@ class URL {
   }
 }
 
+Object.defineProperties(URL.prototype, {
+  toString: {
+    // https://heycam.github.io/webidl/#es-stringifier
+    writable: true,
+    enumerable: true,
+    configurable: true,
+    value: function(options) {
```

Comment:
You can disable the `func-name-matching` for that one line with a comment. 

```js
configurable: true,
// eslint-disable-next-line func-name-matching
value: function toString(options) {
```

There's also `// eslint-disable-line func-name-matching` but since that has to be appended to the offending line, it often results in a line over 80 characters long, which violates a different lint rule. :-D

FWIW, this came up on the ESLint issue tracker two months ago but hasn't moved much since then.  Doesn't look like there was consensus on whether something should be done or not. https://github.com/eslint/eslint/issues/7423

/cc @not-an-aardvark